dockerfiles/verify_packages/installer/utils.sh

```diff
@@ -12,6 +12,16 @@ assert_file_contains() {
     fi
 }
 
+assert_file_not_contains() {
+    output=$(cat ${1})
+    if [ -z "${output##*$2*}" ]; then
```

Comment:
Sorry, what is this bash expansion doing?

Reply:
TL;DR: if the string in $2 is found, it deletes the string itself, plus everything before and after it.

---

It is [substring removal](https://tldp.org/LDP/abs/html/string-manipulation.html#:~:text=expr%20%22%24stringZ%22%20%3A%20%27.*%5C(......%5C)%27%60%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%23%20ABCabc-,Substring%20Removal,-%24%7Bstring%23substring%7D).

> `${string##substring}`
> Deletes longest match of $substring from front of $string.

Substring removal supports the wildcard `*`, meaning anything found there.